### ISSUE TITLE
Can ZenFS support non-ZNS type SSDs？

### ISSUE THREAD

**Issue by benqwu** [westerndigitalcorporation/zenfs#291]
Can ZenFS support non-ZNS type SSDs, and what is the performance impact?

**Comment by yhr** [comment]
Yes, it can be supported, but one would have to implement it and benchmark
it to measure the benefits.

Implementing this interface for conventional drives(somehow faking zone
operations) could work:
https://github.com/westerndigitalcorporation/zenfs/blob/master/fs/zbdlib_zenfs.h

Write amp could potentially be reduced a bit (delta conventional file
systems), but probably the biggest advantage
would be faster synced writes. ZenFS outperforms ext4 and xfs for small
synchronized buffered(e.g. WAL) writes by about 2x.

**Comment by benqwu** [comment]
Thank you for your reply. Does the ZenFS have a plan to support non-ZNS SSDs so that ZenFS can be applied to a wider range of scenarios?

**Comment by yhr** [comment]
I have no plans to implement support for conventional devices, but would of
course happily review any pull requests in this direction :)
Thanks for your interest!
